api/mock_trace_service.proto

```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
```

Comment:
I'm not sure if we're supposed to commit the `.proto` file or the `.pb.go` file, if we're not then please let me know and I'll remove it

Reply:
We definitely want the .proto file checked in.

It _might_ make sense to also check in the .go file to simplify the library for consumers (such that they don't need to have the protobuf toolchain to use this).

If we check in the .go code we will probably also want a presubmit check to ensure that it is up-to-date with .proto.

Reply:
Got it, I think it might be easier for users if we check in the `.pb.go` file too, since it personally took me a while to setup `protoc`. I'll create an issue to add a presubmit check to validate the proto files.